modules/imgproc/src/peronamalik.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include "precomp.hpp"
+#include "opencl_kernels_imgproc.hpp"
+#include "cvconfig.h"
+
+#include <iostream>
+using namespace std;
+
+namespace cv
+{
+    void PeronaMalik(InputArray src, OutputArray dst, double timeStepSize, double k, int noOfTimeSteps, int fluxFunc )
+    {
+        CV_Assert(k != 0);
+        CV_Assert(timeStepSize > 0);
+        CV_Assert(noOfTimeSteps >= 0);
+        CV_Assert((fluxFunc == CV_PERONA_MALIK_EXPONENTIAL) || (fluxFunc == CV_PERONA_MALIK_INVERSE_QUADRATIC));
+        CV_Assert(src.size() == dst.size());
+        CV_Assert(src.type() == dst.type());
```

Comment:
We should not change input data (after `swap(_src, _dst)`), so we need to clone data into new buffer.